src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -979,6 +979,16 @@ public static bool SequenceEqual<TSource>(this IEnumerable<TSource> first, IEnum
             if (comparer == null) comparer = EqualityComparer<TSource>.Default;
             if (first == null) throw Error.ArgumentNull("first");
             if (second == null) throw Error.ArgumentNull("second");
+
+            if (first == second) return true;
```

Comment:
This is a standard enough short-circuit on any O(n) equality check, but I suppose something **could** return different results when enumerated twice. To have that and to test for it being equal seems bizarre, but is it bizarre enough to consider this safe?